src/compiler/crystal/macros/interpreter.cr

```diff
@@ -546,6 +546,8 @@ module Crystal
       when "@type"
         target = @scope == @program.class_type ? @scope : @scope.instance_type
         @last = TypeNode.new(target.devirtualize)
+      when "@top_level"
+        @last = TypeNode.new(@program.devirtualize)
```

Comment:
`devirtualize` isn't needed here, but using it does no harm (it just returns `self` in the case of `@program`)

```suggestion
        @last = TypeNode.new(@program)
```